std/experimental/color/colorspace.d

```diff
@@ -0,0 +1,419 @@
+    DCI_P3_D65,
+}
+
+
+/**
+Chromatic adaptation method.
+*/
+enum ChromaticAdaptationMethod
+{
+    /** Direct method, no correction for cone response. */
+    XYZ,
+    /** Bradford method. Considered by most experts to be the best. */
+    Bradford,
+    /** Von Kries method. */
+    VonKries
+};
+
+
+/**
+Parameters that define an RGB color space.
+*/
+struct RGBColorSpaceDesc(F)
+{
+    /** Gamma conversion function type. */
+    alias GammaFunc = F function(F v) pure nothrow @nogc @safe;
+
+    /** Color space name. */
+    string name;
+
+    /** Function that converts a linear luminance to gamme space. */
+    GammaFunc toGamma;
+    /** Function that converts a gamma luminance to linear space. */
+    GammaFunc toLinear;
+
+    /** White point. */
+    xyY!F white;
+    /** Red point. */
+    xyY!F red;
+    /** Green point. */
+    xyY!F green;
+    /** Blue point. */
+    xyY!F blue;
+}
+
+/**
+Color space descriptor for the specified color space.
+*/
+RGBColorSpaceDesc!F rgbColorSpaceDef(F = double)(RGBColorSpace colorSpace) if(isFloatingPoint!F)
+{
+    return rgbColorSpaceDefs!F[colorSpace];
+}
+
+/**
+RGB to XYZ color space transformation matrix.
+*/
+F[3][3] rgbToXyzMatrix(F = double)(RGBColorSpaceDesc!F cs) if(isFloatingPoint!F)
+{
+    static XYZ!F toXYZ(xyY!F c) { return c.y == F(0) ? XYZ!F() : XYZ!F(c.x/c.y, F(1), (F(1)-c.x-c.y)/c.y); }
+
+    // build a matrix from the 3 color vectors
+    auto r = toXYZ(cs.red);
+    auto g = toXYZ(cs.green);
+    auto b = toXYZ(cs.blue);
+    F[3][3] m = [[ r.X, g.X, b.X],
+                 [ r.Y, g.Y, b.Y],
+                 [ r.Z, g.Z, b.Z]];
+
+    // multiply by the whitepoint
+    F[3] w = [ toXYZ(cs.white).tupleof ];
+    auto s = multiply(inverse(m), w);
+
+    // return colorspace matrix (RGB -> XYZ)
+    return [[ r.X*s[0], g.X*s[1], b.X*s[2] ],
+            [ r.Y*s[0], g.Y*s[1], b.Y*s[2] ],
+            [ r.Z*s[0], g.Z*s[1], b.Z*s[2] ]];
+}
+
+/**
+XYZ to RGB color space transformation matrix.
+*/
+F[3][3] xyzToRgbMatrix(F = double)(RGBColorSpaceDesc!F cs) if(isFloatingPoint!F)
+{
+    return inverse(rgbToXyzMatrix(cs));
+}
+
+/**
+Generate a chromatic adaptation matrix from srcWhite to destWhite.
+*/
+F[3][3] chromaticAdaptationMatrix(ChromaticAdaptationMethod method = ChromaticAdaptationMethod.Bradford, F = double)(xyY!F srcWhite, xyY!F destWhite) if(isFloatingPoint!F)
+{
+    enum Ma = chromaticAdaptationMatrices!F[method];
+    enum iMa = inverse!F(Ma);
+    auto XYZs = convertColor!(XYZ!F)(srcWhite);
+    auto XYZd = convertColor!(XYZ!F)(destWhite);
+    F[3] Ws = [ XYZs.X, XYZs.Y, XYZs.Z ];
+    F[3] Wd = [ XYZd.X, XYZd.Y, XYZd.Z ];
+    auto s = multiply!F(Ma, Ws);
+    auto d = multiply!F(Ma, Wd);
+    F[3][3] t = [[d[0]/s[0], F(0),      F(0)     ],
+                 [F(0),      d[1]/s[1], F(0)     ],
+                 [F(0),      F(0),      d[2]/s[2]]];
+    return multiply!F(multiply!F(iMa, t), Ma);
+}
+
+/** Linear to hybrid linear-gamma ramp function. */
```

Comment:
> seems dodgey?

Maybe. We already have some modules linking to wikipedia articles though. I'd also add Wikipedia links to `HSx`, `LAB`, `RGB`... to the module documentation as this might be useful for some users. OTOH it's probably a matter of taste.